forecasting/src/autogluon/forecasting/utils/dataset_utils.py

```diff
@@ -0,0 +1,297 @@
+        each time series, and use dataframe.pivot() to convert it to one column, where the column name is the
+        time, each row is the corresponding target value for each time series.
+        """
+        df = df.sort_values(by=index_column)
+        data_dic = {index_column: sorted(list(set(df[index_column])))}
+        # check for dataset with multiple targets for a certain time series/time
+        if any(df[[index_column, time_column]].value_counts() > 1):
+            for combination in df[[index_column, time_column]].value_counts().index:
+                raise ValueError(f"Containing multiple targets for time series {combination[0]} and time {combination[1]}. "
+                                 "Please check your dataset.")
+        # check whether need auto-padding
+        need_padding = False
+        if any(df[time_column].value_counts() != len(df[index_column].unique())):
+            logger.log(30, "Warning: autogluon requires each index to be observed over the same set of time values. \n"
+                           "As this is not the case in your data, we are automatically padding the dataset with all missing (index, time) combinations which may take some time. \n"
+                           "To do this yourself, simply add rows to the Dataframe with target value = NA for each missing (index,time) combination.")
+            need_padding = True
+
+        for time in time_list:
+            tmp = df[df[time_column] == time][[index_column, time_column, target_column]]
+            tmp = tmp.pivot(index=index_column, columns=time_column, values=target_column)
+            # automatically padding with NAN if for some time series missing a date target if needed
+            if need_padding:
+                for index in data_dic[index_column]:
+                    if index not in tmp.index:
+                        tmp.loc[index, time] = None
+            tmp_values = tmp[time].values
+            data_dic[time] = tmp_values
+        return pd.DataFrame(data_dic)
+
+    X = reshape_dataframe(X)
+    return X
+
+# TODO: Improve the way to do the split
+def train_test_split_dataframe(data, prediction_length):
+    test_ds = data.copy()
+    train_ds = data.iloc[:, :-prediction_length]
+    if not any(data.iloc[:, -prediction_length:]):
+        logger.log(30, "Warning: All targets used for validation is NAN.")
+    return train_ds, test_ds
+
+
+def train_test_split_gluonts(data, prediction_length, freq=None):
+    train_data_lst = []
+    test_data_lst = []
+
+    if freq is None:
+        if isinstance(data, ListDataset):
+            freq = data.list_data[0]["start"].freq
+        else:
+            raise ValueError("Cannot infer freq if the data is not ListDataset.")
+
+    for entry in data:
+        test_data_lst.append(entry.copy())
+        tmp = {}
+        for k, v in entry.items():
+            if "dynamic" in k.lower():
+                tmp[k] = v[..., : -prediction_length]
```

Comment:
maybe use more descriptive name than `tmp`

Reply:
+1, also maybe using more descriptive name for  `k` and `v`.